qa/tasks/ceph_salt.py

```diff
@@ -0,0 +1,274 @@
+
+    Assumes a Salt cluster is already running (use the Salt task to achieve
+    this).
+
+    This task understands the following config keys which apply to
+    this task and all its subtasks:
+
+        log_anchor      a string (default: "WWWW: ") which will precede
+                        log messages emitted at key points during the
+                        deployment
+        quiet_salt:
+            true        suppress stderr on salt commands (the default)
+            false       let salt commands spam the log
+        allow_reboots:
+            true        Allow cluster nodes to be rebooted if needed (default)
+            false       
+        deploy:
+            true        Enable role deployment on ceph-salt (default)
+            false       
+    """
+    err_prefix = "(ceph_salt task) "
+
+    log_anchor_str = "WWWW: "
+
+    def __init__(self, ctx, config):
+        super(CephSalt, self).__init__(ctx, config)
+        log.debug("beginning of constructor method")
+        if not ceph_salt_ctx:
+            self._populate_ceph_salt_context()
+            self.log_anchor = ceph_salt_ctx['log_anchor']
+            introspect_roles(self.ctx, self.log, quiet=False)
+            self.ctx['roles'] = self.ctx.config['roles']
+            self.log = log
+        self.reboots_explicitly_forbidden = not self.config.get("allow_reboots", True)
+        self.master_remote = ceph_salt_ctx['master_remote']
+        self.quiet_salt = ceph_salt_ctx['quiet_salt']
+        self.nodes = self.ctx['nodes']
+        self.nodes_storage = self.ctx['nodes_storage']
+        self.nodes_storage_only = self.ctx['nodes_storage_only']
+        self.remotes = self.ctx['remotes']
+        self.roles = self.ctx['roles']
+        self.sm = ceph_salt_ctx['salt_manager_instance']
+        self.role_types = self.ctx['role_types']
+        self.remote_lookup_table = self.ctx['remote_lookup_table']
+        self.ceph_salt_deploy = ceph_salt_ctx['deploy']
+        self.scripts = Scripts(self.ctx, self.log)
+
+    def _install_ceph_salt(self):
+        '''
+        Installs ceph-salt on master either from source if repo and/or branch are 
+        provided in the suite yaml or from rpm if not
+        '''
+        global ceph_salt_ctx
+        if ceph_salt_ctx['repo']:
+            if not ceph_salt_ctx['branch']:
+                self.scripts.run(
+                self.master_remote,
+                'install_ceph_salt.sh',
+                args=ceph_salt_ctx['repo']
+                )
+            else:
+                self.scripts.run(
+                self.master_remote,
+                'install_ceph_salt.sh',
+                args=[ceph_salt_ctx['repo'], ceph_salt_ctx['branch']]
+                )
+        else:
+            self.scripts.run(
+            self.master_remote,
+            'install_ceph_salt.sh'
+            )
+        self.ctx.cluster.run(args=[ 'sudo', 'sh', '-c',
+                                    'systemctl restart salt-minion'])
```

Comment:
'sh', '-c' are not required, you can use just:
self.ctx.cluster.run(args='sudo systemctl restart salt-minion')